apps/aecore/lib/aecore/persistence/worker.ex

```diff
@@ -7,50 +7,124 @@ defmodule Aecore.Persistence.Worker do
   use GenServer
 
   alias Aecore.Chain.BlockValidation
+  alias Aecore.Keys.Worker, as: Keys
 
   require Logger
 
+  @latest_block_key "latest_block"
+
   def start_link(_args) do
     GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
   end
 
   ## Client side
 
-  @spec write_block_by_hash(Block.t()) :: :ok | {:error, reason :: term()}
-  def write_block_by_hash(%{header: header} = block) do
+  @spec add_block_by_hash(Block.t()) :: :ok | {:error, reason :: term()}
+  def add_block_by_hash(%{header: header} = block) do
     hash = BlockValidation.block_header_hash(header)
-    GenServer.call(__MODULE__, {:write_block_by_hash, {hash, block}})
+    GenServer.call(__MODULE__, {:add_block_by_hash, {hash, block}})
+  end
+  def add_block_by_hash(_block), do: {:error, "bad block structure"}
+
+  @spec get_block_by_hash(String.t()) ::
+  {:ok, block :: Block.t()} | :not_found | {:error, reason :: term()}
```

Comment:
can you put the spec in one line, this is like this multiple times

Reply:
I'm trying to keep my lines no longer than 80 columns.